tests/vyos/modules/interfaces/merged.yaml

```diff
@@ -0,0 +1,63 @@
+---
+- debug:
+      msg: "START vyos_interfaces merged integration tests on connection={{ ansible_connection }}"
+
+- include_tasks: remove_config.yaml
+
+- name: Merge the provided configuration with the exisiting running configuration
+  vyos_interfaces: &merged
+      config:
+          - name: eth1
+            description: "Configured by Ansible - Interface 1"
+            mtu: 1500
+            speed: auto
+            duplex: auto
+            vifs:
+                - vlan_id: 100
+                  description: "Eth1 - VIF 100"
+                  mtu: 400
+
+                - vlan_id: 101
+                  description: "Eth1 - VIF 101"
+
+          - name: eth2
+            description: "Configured by Ansible - Interface 2 (ADMIN DOWN)"
+            mtu: 600
+            enable: false
+
+          - name: eth3
+            description: "Configured by Ansible - Interface 3"
+            speed: 100
+            duplex: full
+      state: merged
+  register: result
+
+- name: Assert that before dicts were correctly generated
+  assert:
+      that: "{{ merged['before'] | difference(result['before']) |length<1 }}"
```

Comment:
Would it be cleaner to test with == ? 

```
- hosts: localhost
  gather_facts: false
  vars:
    first:
      some:
        nested:
          key:
            as:
              a:
                boolean: True
    second:
      some:
        nested:
          key:
            as:
              a:
                boolean: True
    third:
      some:
        nested:
          key:
            as:
              a:
                boolean: False
  tasks:
  - assert:
      that: "{{ first == second }}"
  - assert:
      that: "{{ first != third }}"
```

Reply:
@cidrblock so the reason why I took this approach is:

- Doing a 'set' difference between two lists would take care of ordering, i.e., we need not have to worry about how (in which order) the before/after list of dicts and the list of commands were generated. Since we're doing object comparisons, changes in ordering also fails the assertions.

- I _could_ define the desired state of each interface (* each operation) as a dict and then do a `==` check, for example:
```yaml
merged:
  before:
    eth0:
      name: "eth0"
      enable: true

    eth1:
      name: "eth1"
      enable: true

    eth2:
      name: "eth2"
      enable: true 

    eth3:
      name: "eth3"
      enable: true
```
However, that would also need us to iterate through the module output before/after lists and convert it into this format?

Reply:
As discussed, I'll update `difference` to `symmetric_difference`.